### PR TITLE
Introduce warning if the current istioctl version near or after EOL date

### LINKED ISSUE #41747: Warn on `istioctl install` when the version is EOL
Strawman implementation:

* Before we ship a release, guess when we will EOL it (this is generally known, but in theory we can change the EOL so its not perfect)
* In istioctl. add `if today > EOL date { warn }`
* In helm, NOTES.txt can do the same thing

The reason we guess the EOL is because we don't want to do a dynamic internet lookup. Because we might be wrong, we may want to say something like "Warning: may be installing an EOL version: see istio.io/support for supported releases" or something, rather than "You **are** installing a EOL version"

cc @therealmitchconnors

### PR DESCRIPTION
**Pull request by zufardhiyaulhaq**
Signed-off-by: zufardhiyaulhaq <zufardhiyaulhaq@gmail.com>

**Please provide a description of this PR:**
fix https://github.com/istio/istio/issues/41747, introduce a warning if the current date is near or after EOL of the date defined 
in istioctl.

telegram-cloud-photo-size-5-6077778448312546357-y
telegram-cloud-photo-size-5-6077778448312546359-y

### CODE REVIEW

File: operator/cmd/mesh/install.go
Comment (howardjohn) [inline operator/cmd/mesh/install.go:172]: I think we should just always say "MAY". The reason is we could retroactively extend support for a version.

Also the month vs year check is not really stable; we could EOL on Jan 2021 and we would warn "maybe" for 11 months, or on Nov 2021 and we would only warn for 1 month?

File: operator/version/version.go
Comment (howardjohn) [inline operator/version/version.go:27]: @ericvn how can we make sure this is updated?

**Review by howardjohn — COMMENTED** [review]
LGTM overall thanks!

File: operator/cmd/mesh/install.go
Comment (zufardhiyaulhaq) [reply operator/cmd/mesh/install.go:172]: can you elaborate more on

`Also the month vs year check is not really stable; we could EOL on Jan 2021 and we would warn "maybe" for 11 months, or on Nov 2021 and we would only warn for 1 month?`

I don't get the idea

Comment (howardjohn) [reply operator/cmd/mesh/install.go:172]: I assume the intent was that `t.Year() > operatorVer.OperatorEOLYear` check means "You are really far EOL" and the next check is "you maybe JUST became EOL".

if so, its better to check the exact delta (e.g 6 months after expected EOL) instead of just the year.

However, I think we should always say MAY anyways so we don't need the check at all.

File: operator/version/version.go
Comment (howardjohn) [reply operator/version/version.go:27]: We do, just want to make sure we also notice to change this

File: operator/cmd/mesh/install.go
Comment (zufardhiyaulhaq) [reply operator/cmd/mesh/install.go:170]: sure, always say `MAY` is preferred here due to changes of extended support.

File: operator/version/version.go
Comment (zufardhiyaulhaq) [reply operator/version/version.go:27]: I think this should be fine, make it as time.Time() required us to use time.Parse() or time.Date() which add complexity

Comment (ericvn) [reply operator/version/version.go:27]: We can add something into the release-builder code as well as the over-arching issue to make sure this happens as part of the updates.

File: operator/version/version.go
Comment (ericvn) [inline operator/version/version.go:28]: If I look at https://istio.io/latest/docs/releases/supported-releases/ and go back to the release 1 yr ago, it was 1.13 and it's EOL was Oct 12. Thus I would suspect that we don't want to actually output a warning until November of 2023.

Comment (zufardhiyaulhaq) [reply operator/version/version.go:28]: Actually, this is configurable for each release, I am not sure 1.17 when will be EOL, so I just put that date.

**Comment by zufardhiyaulhaq** [comment]
Does anything need to check @ericvn @howardjohn to get this merged?

**Review by therealmitchconnors — APPROVED** [review]
Thanks for your contribution!

**Review by ericvn — APPROVED** [review]
Thanks. Will update the release process to update this in both the new branch as well as the main branch.

**Comment by zufardhiyaulhaq** [comment]
@ericvn @therealmitchconnors can we get this merged?

**Comment by ericvn** [comment]
@zufardhiyaulhaq If you look a the bottom of the PR it says Code owner review required and it's still looking for: `Waiting on code owner review from istio/wg-environments-maintainers.`

@istio/wg-environments-maintainers would you take a look.

**Comment by zufardhiyaulhaq** [comment]
Hi @istio/wg-environments-maintainers can you help take a look?

File: operator/version/version.go
Comment (ericvn) [reply operator/version/version.go:28]: 1.18.0 which is now what the main branch is, will likely EOL in 10/23 so maybe make this November.

**Comment by zufardhiyaulhaq** [comment]
@ericvn @howardjohn @therealmitchconnors is this still needed? I can rebase

**Review by howardjohn — APPROVED** [review]

**Pull request merged**
